js/main.js

```diff
@@ -1,13 +1,53 @@
+const formData = localStorage.getItem('form-data') ? JSON.parse(localStorage.getItem('form-data')) : {
+  Full_name: '',
+  email: '',
+  message: '',
+};
+
+let form;
+
+function loadData(tempFormData) {
+  Object.keys(tempFormData).forEach((inputName) => {
+    form[inputName].value = tempFormData[inputName];
+  });
+}
+
+function saveData(tempFormData) {
+  localStorage.setItem('form-data', JSON.stringify(tempFormData));
+}
+
 function menuToggle() {
   document.querySelector('#nav-menu').classList.toggle('toggle-mobile-menu');
 }
 
 window.addEventListener('load', () => {
+  form = document.getElementById('contact-me-form');
+  loadData(formData);
   document.querySelector('#menu-btn').addEventListener('click', () => {
     menuToggle();
   });
 
   document.querySelector('#nav-menu').addEventListener('click', () => {
     menuToggle();
   });
+
+  form.addEventListener('submit', (event) => {
+    if (document.querySelector('small')) form.removeChild(document.querySelector('small'));
+    if (form.email.value !== form.email.value.toLowerCase()) {
+      event.preventDefault();
+      const listItem = document.createElement('li');
+      const errorMessage = document.createElement('small');
+      errorMessage.innerText = 'Email Contains Capital Letters';
+      listItem.append(errorMessage);
+      form.querySelector('ul').append(listItem);
+    }
+  });
+
+  document.querySelectorAll('input, textarea').forEach((inp) => {
+    inp.addEventListener('change', () => {
+      const inputName = inp.getAttribute('name');
+      formData[inputName] = inp.value;
+      saveData(formData);
+    });
+  });
 });
```

Comment:
## [OPTIONAL]
- [ ] This is not in the requirements is very important. KIndly I would suggest you add your past milestones to this project. Currently, your popup functionality is not working. Kindly make sure you add it to this milestone.

![image](https://user-images.githubusercontent.com/65068771/151439224-0dd2d3e5-6f2d-4d70-b4a6-a9c3b4055bd5.png)